### DISCUSSION TITLE
DynamoDB getting error when creating client

### DISCUSSION THREAD

**Discussion by pgotelliNonerds** [aws/aws-sdk-js-v3#5261] [Q&A]
```{
    "errorType": "TypeError",
    "errorMessage": "(0 , protocol_http_1.getHttpHandlerExtensionConfiguration) is not a function",
    "stack": [
        "TypeError: (0 , protocol_http_1.getHttpHandlerExtensionConfiguration) is not a function",
        "    at resolveRuntimeExtensions (/opt/nodejs/node_modules/@aws-sdk/client-dynamodb/dist-cjs/runtimeExtensions.js:12:79)",
        "    at new DynamoDBClient (/opt/nodejs/node_modules/@aws-sdk/client-dynamodb/dist-cjs/DynamoDBClient.js:33:76)",
        "    at Object.<anonymous> (/var/task/app.js:10:24)",
        "    at Module._compile (node:internal/modules/cjs/loader:1256:14)",
        "    at Module._extensions..js (node:internal/modules/cjs/loader:1310:10)",
        "    at Module.load (node:internal/modules/cjs/loader:1119:32)",
        "    at Module._load (node:internal/modules/cjs/loader:960:12)",
        "    at Module.require (node:internal/modules/cjs/loader:1143:19)",
        "    at require (node:internal/modules/cjs/helpers:121:18)",
        "    at _tryRequireFile (file:///var/runtime/index.mjs:976:37)"
    ]
}

**Comment by yenfryherrerafeliz** [comment]
Hi @pgotelliNonerds, could you please provide a sample code that reproduced your issue, SDK version, and runtime being used?

Thanks!

**Comment by pacosw1** [comment]
Got the same issue, it means your credentials expired or are not setup correctly, I entered aws configure and misclicked and modified the secret key so it was showing as undefined

**Comment by brucruz** [comment]
I am getting a similar error that's coming from RDS:
```
Error: (0 , protocol_http_1.getHttpHandlerExtensionConfiguration) is not a function

Trace: TypeError: (0 , protocol_http_1.getHttpHandlerExtensionConfiguration) is not a function
    at resolveRuntimeExtensions (/Users/brunocruz/code/ribbon/ribbon-app/node_modules/sst/node_modules/@aws-sdk/client-rds-data/dist-cjs/runtimeExtensions.js:12:79)
    at new RDSDataClient (/Users/brunocruz/code/ribbon/ribbon-app/node_modules/sst/node_modules/@aws-sdk/client-rds-data/dist-cjs/RDSDataClient.js:28:76)
    at useAWSClient (file:///Users/brunocruz/code/ribbon/ribbon-app/node_modules/sst/credentials.js:64:20)
    at file:///Users/brunocruz/code/ribbon/ribbon-app/node_modules/sst/cli/commands/plugins/warmer.js:8:20
    at Module.<anonymous> (file:///Users/brunocruz/code/ribbon/ribbon-app/node_modules/sst/util/lazy.js:6:22)
    at file:///Users/brunocruz/code/ribbon/ribbon-app/node_modules/sst/cli/commands/dev.js:336:61
    at async Promise.all (index 9)
    at async Object.handler (file:///Users/brunocruz/code/ribbon/ribbon-app/node_modules/sst/cli/commands/dev.js:322:9)
    at process.<anonymous> (file:///Users/brunocruz/code/ribbon/ribbon-app/node_modules/sst/cli/sst.js:60:21)
    at process.emit (node:events:525:35)
    at process.processEmit [as emit] (/Users/brunocruz/code/ribbon/ribbon-app/node_modules/signal-exit/index.js:199:34)
    at process._fatalException (node:internal/process/execution:149:25)
    at processPromiseRejections (node:internal/process/promises:288:13)
    at process.processTicksAndRejections (node:internal/process/task_queues:96:32)
```

Were you able to fix yours, @pgotelliNonerds ?

**Reply by brucruz** [reply]
I fixed it by removing node_modules and installing it again 🤷‍♂️

**Comment by enisze** [comment]
Worked for me too @brucruz but not on vercel. Any suggestions there?

**Reply by brucruz** [reply]
Sorry, no. I use the AWS cloud environment

**Comment by afarah1** [comment]
🔎 `protocol_http_1` is `require("@smithy/protocol-http")` (from [here](https://github.com/aws/aws-sdk-js-v3/blob/main/clients/client-dynamodb/src/runtimeExtensions.ts#L6)), `@smithy/protocolo-http` declares `getHttpHandlerExtensionConfiguration` at `src/extensions` ([here](https://github.com/smithy-lang/smithy-typescript/blob/f4e0bd9a40c75c7839ec7cb879a467eb1ee34929/packages/protocol-http/src/extensions/httpExtensionConfiguration.ts#L25)), so it should not be `undefined`. However, for me `src/extensions` was not present in the `@smithy/protocol-http` version being used (`1.2.0`). It is present in the latest version (`4.1.4`). A similar mismatch is probably why @brucruz was able to solve it by deleting his `node_modules` and installing again.

Our version mismatch was caused by us using `@aws-sdk/protocol-http` which is deprecated in favor of `@smithy/protocol-http`:

```
$ npm list '@smithy/protocol-http'
...
    ├─┬ @aws-sdk/protocol-http@3.374.0
    │ └── @smithy/protocol-http@1.2.0 
    └── @smithy/protocol-http@4.1.4  extraneous
```

✅ Removing the spurious dependency on `@aws-sdk/protocol-http` from our package solved the issue for us. 

If you are having this issue check if you are using an old version of `@smithy/protocol-http` (go to the file required where the exception occurs and check the version on `package.json`, it may not show on `npm list`). If so, remove or upgrade the old package.

This is most likely NOT an SDK issue.
